feat(webapps): refactor webapps into assembly & frontend modules

danielkelemen opened the pull request:
related to #3098

tasso94 commented:
Shouldn't `webapps/assembly/src/libs/globalize.js` rather be part of `frontend`?

danielkelemen commented:
> Shouldn't `webapps/assembly/src/libs/globalize.js` rather be part of `frontend`?

Yes, it should.

tasso94 reviewed (changes requested):
Looks good to me, except for the comment I added above. However, the distro-ee stage fails.

**Edit:** I think this is because the `assembly/assembly.xml` file needs changes.
**Edit II:** We need to make the vulnerability scanner find the `package.json` and `package-lock.json` again after your change. Cf. https://jira.camunda.com/browse/CAM-14396

tasso94 commented:
I created a PR for the vulnerability scanner https://github.com/camunda/automation-platform-vulnerability-scan/pull/1708 since the solution in CAM-14396 is not functional. With the new config property `scan-directories`, we can add directories separated by a comma.

danielkelemen commented:
Implementation details:
- Create `assembly` module inside the `webapps` and move all the java code into this.
  - I decided to not add a root pom in the webapps yet so we can keep the history for the `pom.xml` in the `assembly` module.
  - It will be added with the Jakarta module in the next step.
- Updated references where needed (CI, frontend, assembly)
- Added jaxb & graal dependencies, needed for higher JDK versions.
- Updated README

tasso94 reviewed (approved):
Looks good to me.

danielkelemen commented:
I will rebase and merge this after the alpha code freeze.
